src/pages/en/reference/api-reference.md

```diff
@@ -158,6 +158,29 @@ You could pass a callback function that renders our the message:
 <div>Hello world!</div>
 ```
 
+### `Astro.self`
+
+`Astro.self` allows Astro components to be recursively called. This behaviour lets you call the same Astro component over again by defining `<Astro.self>` within the component template.  Allowing you to easily iterate over large data stores and nested data-structures.
+
+```astro
+---
+const {frontmatter, nestedHeaders} = Astro.props 
+const Headers = frontmatter.headers || nestedHeaders
```

Comment:
It might be useful for the example to simplify this. Or even better can we just inline the data we're thinking about so that it's easy to see?

So declare some recursive data in the front matter and then use the template Astro.self to render it?

Oh wait. Or we can't do that can we? Because the self needs to receive props. In that case maybe show the component in a separate code block being passed the data?